api/envoy/api/v2/listener/listener.proto

```diff
@@ -121,6 +121,16 @@ message FilterChainMatch {
   // * ``tls`` - set by :ref:`envoy.listener.tls_inspector <config_listener_filters_tls_inspector>`
   //   when TLS protocol is detected.
   string transport_protocol = 9;
+
+  // If non-empty, a list of application protocols to consider when determining a filter chain
+  // match. Those values will be compared against the application protocols of a new connection,
+  // when detected by one of the listener filters (this needs to be configured separately).
+  //
+  // Valid values include:
+  //
+  // * ``http/1.1`` - set by :ref:`envoy.listener.tls_inspector <config_listener_filters_tls_inspector>`,
```

Comment:
In the common case, if trying to split h2 vs h1 traffic, would it make sense to have a match for 'h2' and a match for 'http/1.1', or would it make more sense to have 'h2' and <empty>? I'm guessing that many older clients that don't support h2 at all won't set any ALPN value. Do you think some commentary about this should be in the documentation here? (I don't have much experience with ALPN, so I'm not sure what the best-practice is here, or even if my question makes a lot of sense).

Reply:
It makes sense, I'll add it to the doc.

I've been working in this space for so long, that most of this is obvious to me and I keep forgetting that's not universally true. Thanks for pointing this out!

Reply:
Note added.